scripts/check-ppxlib-floor.sh

```
@@ -0,0 +1,145 @@
+  { note "check-ppxlib-floor: expected exactly one ppxlib bound in $DUNE_PROJECT, found $(printf '%s\n' "$dp_bound" | grep -c .)"; exit 2; }
+[ "$(printf '%s\n' "$op_bound" | grep -c .)" = 1 ] ||
+  { note "check-ppxlib-floor: expected exactly one ppxlib bound in $OPAM_FILE, found $(printf '%s\n' "$op_bound" | grep -c .)"; exit 2; }
+
+if [ "$dp_bound" != "$op_bound" ]; then
+  bad "$DUNE_PROJECT declares ppxlib >= $dp_bound but $OPAM_FILE publishes >= $op_bound."
+  bad "  $OPAM_FILE is generated from $DUNE_PROJECT AND tracked, and it is the one"
+  bad "  opam clients read. Regenerate it, or apply the same line by hand."
+fi
+
+# --- rule 2: the KB records the same floor ----------------------------------
+kb_bound=$(sed -nE 's/.*[Dd]eclared ppxlib floor: `([0-9][0-9.]*)`.*/\1/p' "$KB_FILE" | head -1)
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)